src/graph/heterograph.cc

```diff
@@ -277,6 +277,29 @@ HeteroGraphPtr HeteroGraph::CopyTo(HeteroGraphPtr g, const DGLContext& ctx) {
       hgindex->meta_graph_, rel_graphs, hgindex->num_verts_per_type_));
 }
 
+HeteroGraphPtr HeteroGraph::PinMemory(HeteroGraphPtr g) {
+  bool allpinned = true;
+  auto hgindex = std::dynamic_pointer_cast<HeteroGraph>(g);
+  CHECK_NOTNULL(hgindex);
+  auto rel_g = hgindex->relation_graphs_;
+
+  auto it = std::find_if_not(
+      rel_g.begin(), rel_g.end(), [](auto& subg) { return subg->IsPinned(); });
+  // all pinned, return itself.
```

Comment:
Every comment should be a sentence.
`// All underlying relation graphs are pinned, return the input hetero-graph directly.`